File: longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        strs.sort(key = len)
        length = len(strs[0])
        found = False
        layer_2 = False
        for i in range(length):
            for j in range(1, len(strs)):
                if strs[0][i] != strs[j][i]:
                    #print("found", i)
                    found = True
                if found == True:
                    layer_2 = True
            if layer_2 == True:
                break
        if found == False:
            return(strs[0])
        elif i > 0:
            return(strs[0][:i])
        else:
            return("")

File: longest-common-prefix/longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        String res = "";
        Arrays.sort(strs, (a, b)-> a.length() - b.length());
        for(int i = 0; i < strs[0].length(); i++) {
            String temp = strs[0].substring(0, i + 1);
            int count = 0;
            for (int j = 1; j < strs.length; j++) {
                for (int end = 0; end < strs[j].length(); end++) {
                    String curr = strs[j].substring(0, end + 1);
                    //System.out.println(temp + " " + curr);
                    if (temp.equals(curr)) {
                        count++;
                    }
                }
            }
            if (count == strs.length - 1) {
                res = temp;
            }
        }
        return res;
    }
}